### PR TITLE
[SYCLomatic] Adding arg_index_input_iterator + key_value_pair tests

### PR DESCRIPTION
**Pull request by danhoeflinger**
Adding tests for dpct::arg_index_input_iterator and dpct::key_value_pair.

Associated changes for this PR in SYCLomatic:
https://github.com/oneapi-src/SYCLomatic/pull/332

### CODE REVIEW

**Comment by danhoeflinger** [comment]
This cannot pass testing until https://github.com/oneapi-src/SYCLomatic/pull/332 is merged, as it is testing new functionality introduced there.

File: help_function/src/onedpl_test_arg_index_input_iterator.cpp [hunk 1]
```diff
@@ -0,0 +1,502 @@
+// ====------ onedpl_test_arg_index_input_iterator.cpp---------- -*- C++ -*
+// ----===////
```
Comment (tangjj11) [inline help_function/src/onedpl_test_arg_index_input_iterator.cpp:2]: why change a line.

**Review by tangjj11 — APPROVED** [review]

**Review by ShengchenJ — APPROVED** [review]

File: help_function/src/onedpl_test_arg_index_input_iterator.cpp (hunk 1 above)
Comment (danhoeflinger) [reply help_function/src/onedpl_test_arg_index_input_iterator.cpp:2]: Ah, this is a mistake.  I'll correct it.  Result of clang-format.

Comment (danhoeflinger) [reply help_function/src/onedpl_test_arg_index_input_iterator.cpp:2]: Whoops, it seems it was already merged...
@tangjj11 @ShengchenJ would you like me to open another PR to correct this?

**Pull request merged**